torch/library.h

```diff
@@ -592,12 +594,17 @@ class TORCH_API Library final {
   ///   m.def("add(Tensor self, Tensor other) -> Tensor");
   /// }
   /// ```
+  // template <typename Schema>
+  // Library& def(Schema&& raw_schema, ) & {
+  //   c10::FunctionSchema s = schema(std::forward<Schema>(raw_schema));
+  //   return _def(std::move(s));
+  // }
+
   template <typename Schema>
-  Library& def(Schema&& raw_schema) & {
+  Library& def(Schema&& raw_schema, std::unordered_set<Tags> tags = {}) & {
```

Comment:
couldn't make it optional because the compiler gets confused with the def below. So, maybe tags should be made non optional at other places too